apps/language_server/lib/language_server/providers/formatting.ex

```diff
@@ -53,21 +53,49 @@ defmodule ElixirLS.LanguageServer.Providers.Formatting do
 
   defp can_format?(_uri, _project_dir), do: false
 
-  def should_format?(file_uri, project_dir, inputs) when is_list(inputs) do
+  defp should_format?(file_uri, project_dir, inputs) when is_list(inputs) do
     file_path = file_uri |> SourceFile.abs_path_from_uri()
 
-    inputs
-    |> Stream.flat_map(fn glob ->
-      [
-        Path.join([project_dir, glob]),
-        Path.join([project_dir, "apps", "*", glob])
-      ]
-    end)
-    |> Stream.flat_map(&Path.wildcard(&1, match_dot: true))
-    |> Enum.any?(&(file_path == &1))
+    if File.exists?(file_path) do
+      formatter_dir = find_formatter_dir(project_dir, file_path)
+
+      inputs
+      |> Stream.flat_map(&Path.wildcard(Path.join([formatter_dir, &1]), match_dot: true))
+      |> Enum.any?(&(file_path == &1))
+    else
+      # If the file doesn't actually exist on disk, we can't currently check
```

Comment:
I think this comment is wrong. un unsaved file would have `unsaved:` or `unknown:` schema instead of `file:`. Calling `file_uri |> SourceFile.abs_path_from_uri()` would crash

Reply:
Sounds good -- that was my comment above, I wasn't sure if we needed this but had added to make a test pass. I'll create the file explicitly in the test instead.